emcc.py

```diff
@@ -1195,9 +1186,22 @@ def get_last_setting_change(setting):
       # we also do not support standalone mode in fastcomp.
       shared.Settings.STANDALONE_WASM = 1
 
-    if options.no_entry or ('_main' not in shared.Settings.EXPORTED_FUNCTIONS and
-                            '__start' not in shared.Settings.EXPORTED_FUNCTIONS):
+    if options.no_entry:
       shared.Settings.EXPECT_MAIN = 0
+    elif shared.Settings.STANDALONE_WASM:
+      if '_main' in shared.Settings.EXPORTED_FUNCTIONS:
+        diagnostics.warning('export-main', 'including `_main` in EXPORTED_FUNCTIONS is not necessary in standalone mode')
```

Comment:
this warning feels a little excessive to me. is there any harm in having main there? if not, then maybe better to not warn, so it's less jarring to switch between standalone mode and not.